### ISSUE TITLE
Qanal not using it's own non-daemon threads

### ISSUE THREAD

**Issue by DayoOliyide** [samsara/samsara#55]
This is issue is mainly to share some knowledge, as the solution turned out to be simple.

## Background
When updating timbre from `4.0.2` to `4.5.1`, I noticed that Qanal kept shutting down immediately
after starting, with no errors or exceptions.

## Issue
The actual problem is that Qanal doesn't use any non-daemon threads and the JVM will shut down
when the main thread stops and there are no non-daemon threads running.
Up until Timbre `4.1.5`, a non-daemon thread was being [created for a short time](https://github.com/ptaoussanis/timbre/blob/v4.1.5/src/taoensso/timbre.cljx#L552) when Qanal used 
the [default-output-fn](https://github.com/ptaoussanis/timbre/blob/v4.1.5/src/taoensso/timbre.cljx#L43).
This short lived thread carried out the task of resolving the hostname and then lived for [60 seconds](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#newCachedThreadPool()), which was long enough for other non-daemon threads to be created by zookeeper/kafka connections. Thus it was a combination of this short lived thread and the long living zookeeper/kafka threads that have been keeping Qanal up and running all this time!!!.

After updating to Timbre `4.5.1`, which contains a [commit](https://github.com/ptaoussanis/timbre/commit/d59d90261119cbe5aa57123aa0820d913b479bf6) that replaced the short live thread (that lives for 60secs) with one that dies immediately after hostname resolution. Qanal would die, because there wasn't any non-daemon thread to keep it up and running until the zookeeper/kafka non-daemon threads took over.
The threads that Qanal uses for processing each partition `(async/thread)` actually are **daemon** threads created via [`thread-macro-executor`](https://github.com/clojure/core.async/blob/master/src/main/clojure/clojure/core/async.clj#L435) and [`conc/counted-thread-factory`](https://github.com/clojure/core.async/blob/master/src/main/clojure/clojure/core/async.clj#L426).

**TL;DR**
Qanal's own partition consuming threads are daemon and Qanal inadvertently depends on the non-daemon threads in other libraries to keep it up and running.

## Solution
Use the `future` function call to ensure that each partition consuming thread is non-daemon.
This way we don't accidentally depend on other libraries, as their internal implementation is likely to change.

Hope you find this bit of information useful, as I definitely enjoyed finding/learning this.
[Small useful functions](https://gist.github.com/DayoOliyide/f353b15563675120e408b6b9f504628a#file-thread-info-clj)
